server/item/boots.go

```diff
@@ -60,5 +60,23 @@ func (b Boots) Boots() bool {
 
 // EncodeItem ...
 func (b Boots) EncodeItem() (name string, meta int16) {
-	return "minecraft:" + b.Tier.Name + "_boots", 0
+	return "minecraft:" + b.Tier.Name() + "_boots", 0
+}
+
+// DecodeNBT ...
+func (b Boots) DecodeNBT(data map[string]any) any {
+	if t, ok := b.Tier.(ArmourTierLeather); ok {
+		if v, ok := data["customColor"].(int32); ok {
+			t.Colour = rgbaFromInt32(v)
```

Comment:
Does this actually work? `b.Tier` is not stored as a pointer so I doubt this actually propagates properly. You'd want to run something like `b.Tier = ArmourTierLeather{Colour: rgbaFromInt32(v)}` I think.